appdaemontestframework/hass_mocks.py

```diff
@@ -136,23 +135,66 @@ def _log_log(msg, level='INFO'):
 
 
 class MockHandler:
-    """A class for generating a mock in an object and holding on to info about it.
+    """
+    A class for generating a mock in an object and holding on to info about it.
     :param object_to_patch: The object to patch
-    :param function_name: the name of the function to patch in the object
-    :param side_effect: side effect method to call. If not set, it will just return `None`
-    :param autospec: If `True` will autospec the Mock signature. Useful for getting `self` in side effects.
+    :param function_or_field_name: the name of the function to patch in the
+    object
+    :param side_effect: side effect method to call. If not set, it will just
+    return `None`
+    :param autospec: If `True` will autospec the Mock signature. Useful for
+    getting `self` in side effects.
     """
-    def __init__(self, object_to_patch, function_name, side_effect=None, autospec=False):
-        self.function_name = function_name
-        return_value = None
-        self.patch = mock.patch.object(object_to_patch, self.function_name, create=True,
-                                       autospec=autospec, side_effect=side_effect, return_value=None)
+
+    def __init__(self,
+                 object_to_patch,
+                 function_or_field_name,
+                 side_effect=None,
+                 autospec=False):
+        self.function_or_field_name = function_or_field_name
+        patch_kwargs = self._patch_kwargs(side_effect, autospec)
+        self.patch = mock.patch.object(
+                object_to_patch,
+                function_or_field_name,
+                **patch_kwargs
+        )
         self.mock = self.patch.start()
 
+    def _patch_kwargs(self, side_effect, autospec):
+        return {
+            'create': True,
+            'side_effect': side_effect,
+            'return_value': None,
+            'autospec': autospec
+        }
+
+
+class DictMockHandler(MockHandler):
+    class MockDict(dict):
+        def reset_mock(self):
+            pass
+
+    def __init__(self, object_to_patch, field_name):
+        super().__init__(object_to_patch, field_name)
+
+    def _patch_kwargs(self, _side_effect, _autospec):
+        return {
+            'create': True,
+            'new': self.MockDict()
+        }
+
+
+class SpyMockHandler(MockHandler):
```

Comment:
Much better name for this 👍

Reply:
😊

Reply:
Wait I just realized ... we don't actually use it anywhere, do we? 😅 (`SpyMockHandler`)

It's probably a leftover from when you split the large PR? If so it's cool, let's keep it until you use it, and if not, delete it then 🙂